src/java/org/apache/cassandra/index/sai/StorageAttachedIndexGroup.java

```diff
@@ -353,7 +353,7 @@ public boolean validateSSTableAttachedIndexes(Collection<SSTableReader> sstables
 
             if (indexDescriptor.isPerSSTableIndexBuildComplete())
             {
-                indexDescriptor.checksumPerSSTableComponents();
+                indexDescriptor.checksumPerSSTableComponents(validateChecksum);
 
                 for (StorageAttachedIndex index : indexes)
                 {
```

Comment:
We should pass `validateChecksum` to `checksumPerIndexComponents()` (which should probably be renamed or merged w/ the other validation method, see https://github.com/apache/cassandra/pull/3029/files#r1446766349) as well

Reply:
I passed it there too.